external/stdk/Kconfig

```diff
@@ -116,6 +116,12 @@ config STDK_IOT_CORE_CRYPTO_SUPPORT_ED25519
 	---help---
 		Support Ed25519 based sign, encryption functions.
 
+config MBEDTLS_USER_FILE
```

Comment:
This config is used in mbedtls, but definition is in stdk. It loosk strange.

Reply:
Except the change in `external/openssl/Makefile`, it is not `add deppath for subdir`.

Reply:
for 1# the reason is that it's the USER CONFIG for STDK module, which needs to use mbedtls.
for 2#, here is the consequent modification for adding deppath for openssl, because actually openssl is a wrapper impl. based on mbedtls, so mbedtls/config.h is also included in openssl, and during make depend for openssl, it needs to parse this including file path.

Reply:
how about make 2 commits in this PR ? each for deppath and mbedtls user config seperately

Reply:
I search the whole TizenRT, no other module need to tune mbedtls  default configuration when using mbedtls. so it is new situation for us.

generally, we prefer to modify our module(STDK) instead of existing module (like medtls). 
but as you said, it looks strange that STDK module includes other modules configuration and Kconfig(MBEDTLS_USER_FILE) name.

so I suggest directly add Kconfig and customed config file( in external/include/mbedtls)
in  mbedtls module,  anyway, it is new issue for us, it need to fix reasonablely.